Add method to retrieve the checksum of the end page for a `Packet`

AlexTMjugador opened the pull request:
The Ogg specification demands that each Ogg page has a checksum of its data for error detection and correction purposes. Logically, this checksum is an implementation detail of the Ogg transport, and higher-level decoders need not know it to parse logical bitstreams. However, specialized decoders may want to efficiently use it to compute a checksum of the data that flows through it, avoiding hashing packet data again ([this is my concrete use case for that](https://github.com/OptiVorbis/OptiVorbis/commit/81db11771c082af647d9f791ecdf3ddac9c4a549)). In addition, testing code may also find use in a checksum to assert that the data of the page a packet belongs to is reasonably similar to what's expected.

Given that the Ogg parsing code already reads and compares page checksums, exposing them in the public `Packet` API, which already allows users to retrieve Ogg encapsulation data such as the granule position, comes at practically no additional cost.

AlexTMjugador committed:
Add method to retrieve the checksum of the end page for a `Packet`

The Ogg specification demands that each Ogg page has a checksum of its
data for error detection and correction purposes. Logically, this
checksum is an implementation detail of the Ogg transport, and
higher-level decoders need not know it to parse logical bitstreams.
However, specialized decoders may want to efficiently use it to compute
a checksum of the data that flows through it, avoiding hashing packet
data again. In addition, testing code may also find use in a checksum to
asser that the data of the page a packet belongs to is reasonably
similar to what's expected.

Given that the Ogg parsing code already reads and compares page
checksums, exposing them in the public `Packet` API, which already
allows users to retrieve Ogg encapsulation data such as the granule
position, comes at practically no additional cost.

AlexTMjugador commented:
It looks like a CI checks fail due to a transitive dependency on `socket2` expecting a higher MSRV than 1.56.1. Please let me know if I should bother with this before merging this PR.

est31 commented:
Yeah ideally one would add sth like `cargo update -p socket2 --precise <last version still working>` to the CI script.

AlexTMjugador committed:
Fix build failing on MSRV

`tokio` pulled a dependency on a version of `socket2` that had a too
high MSRV. Because the `tokio` `socket2` version dependency range starts
at a too new `socket2` version, it's necessary to downgrade `tokio` to
the last version that has a MSRV compatible with ours.

AlexTMjugador commented:
CI is green now! :tada:

est31 reviewed (approved):
Thanks!
